### ISSUE TITLE
Version 0.34.2 adds empty dd.env and dd.version values to logs

### ISSUE THREAD

**Issue by stevenbuccini** [DataDog/dd-trace-rb#1003]
We upgraded to `0.34.2` yesterday and are seeing empty `dd.env` and `dd.version` values set in our telemetry.

An example log is here:
```
[dd.trace_id=1418951474465141154 dd.span_id=7662676986696068047 dd.env= dd.version=] 
<log content>
```

The addition of these values broke our grok processors. We do not have `dd.env` or `dd.version` environment variables, nor are we setting it in the tracer config. We are setting the env via tags like so:
`DD_TAGS: "service:api,env:<%= env %>"`.

**Comment by delner** [comment]
Hey @stevenbuccini, this is a feature introduced in 0.34.0, and is required for trace correlation with logs. By design its meant to emit empty values like that for the log processor, if no values are configured.

The introduction of `DD_SERVICE`, `DD_ENV` and `DD_VERSION` replace and override the respective tags in `DD_TAGS`, and we recommend their use instead.

If the empty the `dd.env=` format is problematic, it's something we'll have to address with our logs processing intake, which is out of scope of `dd-trace-rb`. I can bring it up with the team.

**Comment by delner** [comment]
@stevenbuccini Talked to the team and we're going to omit the tags which have blank values. Look for that change in an upcoming release.

**Comment by delner** [comment]
We've merged #1008 to address this issue.
